### DISCUSSION TITLE
Spell checking highlight doesn't work after version 0.10.0

### DISCUSSION THREAD

**Discussion by cpcopy** [neovim/neovim#30758] [Q&A] [answered]
With `[s`, it can move between error words, but the highlight for error words doesn't work. Tried to comment all configuration, it still doesn't work.  Does anyone know what the problem is?
The highlight configurations look fine.
```
SpellBad       xxx cterm=undercurl gui=undercurl guisp=#dc322f
SpellCap       xxx cterm=undercurl gui=undercurl guisp=#6c71c4
SpellRare      xxx cterm=undercurl gui=undercurl guisp=#2aa198 
SpellLocal     xxx cterm=undercurl gui=undercurl guisp=#b58900 
```
Version 0.9.5 works fine.

**Comment by clason** [comment]
First step is always to try `nvim --clean`. (And note that the -- new -- colorscheme doesn't set any colors but only underlines; make sure your terminal can show them.)

**Reply by cpcopy** [reply]
Thanks for commenting.
I use windows terminal. It supports underlines. The example gets from version 0.9.5 without --clean.
![image](https://github.com/user-attachments/assets/2cbba5de-8e75-44cb-9510-a4b6638ac774)
And this is its highlight configuration
![image](https://github.com/user-attachments/assets/06d02b82-8d40-49e2-a2eb-b3d39cef2597)


Following is the highlight configuration of version 0.10.2 with --clean.
![image](https://github.com/user-attachments/assets/ed3849dc-62f7-4508-b650-10ca9b381a8f)

But it still doesn't work.
![image](https://github.com/user-attachments/assets/d864b2a6-fd67-433e-80bf-66915e89e20f)

**Reply by clason** [reply]
Under**curl** is different from under**line**.

**Reply by cpcopy** [reply]
> And note that the -- new -- colorscheme doesn't set any colors but only underlines; 
I don't pretty understand this. Do you mean the current new colorscheme on 0.10.2 only support underlines?
> Undercurl is different from underline.
As the default highlight configuration is undercurl, would it show the undercurl same as what 0.9.5 does?

**Reply by clason** [reply, marked as answer]
Your 0.9.5 screenshot shows underlines, not undercurl. This is what undercurls should look like:
<img width="631" alt="Screenshot 2024-10-12 at 13 31 31" src="https://github.com/user-attachments/assets/4d9d9289-8f02-4346-9579-4ad9252f4575">

And you need to compare `--clean` with `--clean`, otherwise it's Apples and Oranges. If you set these highlight groups yourself, make sure to set `gui=underline`, not `undercurl`.

Another thing that has changed in 0.10 is automatic `termguicolors` in terminals that support truecolor. (This is documented in `:h news`, which is required reading on any update.) So check `:set tgc?` in both cases.

**Reply by cpcopy** [reply]
Thanks detailed explanation. 
Now, I find that the windows terminal version I used doesn't support undercurl. After updating to the version has undercurl supported, issue is resolved.
